src/fmu/dataio/providers/objectdata/_base.py

```diff
@@ -175,83 +73,106 @@ class ObjectDataProvider(Provider):
+                f"The standard folder name is overrided from {obj_data.efolder} to "
                 f"{self.dataio.forcefolder}"
             )
-            objres.efolder = self.dataio.forcefolder
+            self.efolder = self.dataio.forcefolder
             logger.info(msg)
             warn(msg, UserWarning)
 
-        self.metadata["name"] = namedstratigraphy.name
-        self.metadata["stratigraphic"] = namedstratigraphy.stratigraphic
-        self.metadata["offset"] = namedstratigraphy.offset
-        self.metadata["alias"] = namedstratigraphy.alias
-        self.metadata["top"] = namedstratigraphy.top
-        self.metadata["base"] = namedstratigraphy.base
+        self._metadata["name"] = self.name
+        self._metadata["stratigraphic"] = named_stratigraphy.stratigraphic
+        self._metadata["offset"] = named_stratigraphy.offset
+        self._metadata["alias"] = named_stratigraphy.alias
+        self._metadata["top"] = named_stratigraphy.top
+        self._metadata["base"] = named_stratigraphy.base
 
-        self.metadata["content"] = (usecontent := content_model.content)
+        self._metadata["content"] = (usecontent := content_model.content)
         if content_model.content_incl_specific:
-            self.metadata[usecontent] = getattr(
+            self._metadata[usecontent] = getattr(
                 content_model.content_incl_specific, usecontent, None
             )
 
-        self.metadata["tagname"] = self.dataio.tagname
-        self.metadata["format"] = objres.fmt
-        self.metadata["layout"] = objres.layout
-        self.metadata["unit"] = self.dataio.unit or ""
-        self.metadata["vertical_domain"] = list(self.dataio.vertical_domain.keys())[0]
-        self.metadata["depth_reference"] = list(self.dataio.vertical_domain.values())[0]
-        self.metadata["spec"] = objres.spec
-        self.metadata["bbox"] = objres.bbox
-        self.metadata["table_index"] = objres.table_index
-        self.metadata["undef_is_zero"] = self.dataio.undef_is_zero
+        self._metadata["tagname"] = self.dataio.tagname
+        self._metadata["format"] = self.fmt
+        self._metadata["layout"] = obj_data.layout
+        self._metadata["unit"] = self.dataio.unit or ""
+        self._metadata["vertical_domain"] = list(self.dataio.vertical_domain.keys())[0]
+        self._metadata["depth_reference"] = list(self.dataio.vertical_domain.values())[0]
+
+        self._metadata["spec"] = (
+            spec.model_dump(mode="json", exclude_none=True)
+            if (spec := self.get_spec())
+            else None
+        )
+        self._metadata["bbox"] = (
+            bbox.model_dump(mode="json", exclude_none=True)
+            if (bbox := self.get_bbox())
+            else None
+        )
+        self._metadata["time"] = (
+            timedata.model_dump(mode="json", exclude_none=True)
+            if (timedata := self._get_timedata())
+            else None
+        )
```

Comment:
This conditional model dumping will become unnecessary when `self._metadata` is actually and instance of `AnyContent`/`UnsetAnyContent` in a later PR.